### PR TITLE
Add `req` function for validating required inputs/values

### PR DESCRIPTION
**Pull request by jcheng5**

### CODE REVIEW

**Comment by jcheng5** [comment]
I'm having second thoughts about this implementation. I didn't bother with non-standard evaluation for the `...` which means there's no short-circuiting. But it might be cool to be able to do:

```r
req(input$a, !is.na(input$a))
```

which will currently generate a warning.

**Comment by jcheng5** [comment]
OK, should be good to go now.

File: R/utils.R [hunk 1]
```diff
@@ -980,6 +980,105 @@ need <- function(expr, message = paste(label, "must be provided"), label) {
...
+}
+
+# Execute a function against each element of ..., but only evaluate each element
+# after the previous element has been passed to fun_. The return value of fun_
+# is discarded, and only invisible() is returned from dotloop.
+#
+# Can be used to facilitate short-circuit eval on dots.
+dotloop <- function(fun_, ...) {
+  for (i in 1:(nargs()-1)) {
+    fun_(eval(as.symbol(paste0("..", i)), environment()))
```
Comment (wch) [inline R/utils.R]: I think you can leave out the `environment()` here.

**Comment by wch** [comment]
Looks good!

**Pull request merged**